relayer/src/foreign_client.rs

```diff
@@ -787,17 +787,17 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
             Ok(res) => {
                 // If elapsed < refresh_window for the client, `try_refresh()` will
                 // be successful with an empty vector.
-                if let Some(ibc_events) = res.clone() {
+                if let Some(ibc_events) = &res {
                     // The assumption is that only one IbcEventType::ChainError will be
                     // in the resulting Vec<IbcEvent> if an error occurred.
                     match ibc_events
-                        .into_iter()
-                        .find(|e| e.event_type() == IbcEventType::ChainError)
+                        .iter()
+                        .find(|&e| e.event_type() == IbcEventType::ChainError)
```

Comment:
This iterator could again take ownership:

```suggestion
                        .into_iter()
                        .find(|e| e.event_type() == IbcEventType::ChainError)
```

Reply:
I don't think we want to take ownership here since we want to be able to keep using the events.